pkg/controller/node/nodecontroller.go

```diff
@@ -534,37 +539,104 @@ func (nc *NodeController) monitorNodeStatus() error {
 			}
 		}
 	}
+	nc.handleDisruption(zoneToNodeConditions, nodes)
 
+	return nil
+}
+
+func (nc *NodeController) handleDisruption(zoneToNodeConditions map[string][]*api.NodeCondition, nodes *api.NodeList) {
+	newZoneStates := map[string]zoneState{}
+	allAreFullyDisrupted := true
 	for k, v := range zoneToNodeConditions {
 		newState := nc.computeZoneStateFunc(v)
-		if newState == nc.zoneStates[k] {
-			continue
+		if newState != stateFullDisruption {
+			allAreFullyDisrupted = false
 		}
-		if newState == stateFullSegmentation {
-			glog.V(2).Infof("NodeController is entering network segmentation mode in zone %v.", k)
-		} else if newState == stateNormal {
-			glog.V(2).Infof("NodeController exited network segmentation mode in zone %v.", k)
+		newZoneStates[k] = newState
+		if _, had := nc.zoneStates[k]; !had {
+			nc.zoneStates[k] = stateInitial
```

Comment:
I don't think we need to. This "Initial" state is only a temporary previous one. It will assure that we won't have `allWasFullyDisrupted == true`, so we'll always enter the main if stmt. In it we'll either enter 'allAreFullyDisruptedBranch' and set the rate limiter there, or an ordinary loop, which will also set the limiter, as new state will be different than "Initial".